File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-contract/src/test/java/de/invesdwin/scripting/haskell/runtime/contract/callback/ParametersAndReturnsTestDecimal.hs
println("getDecimal")
if isdefined(Main, :getDecimal) && !isnothing(getDecimal)
	error("getDecimal already defined!")
end
getDecimal = callback("getDecimal")
println(typeof(getDecimal))
println(getDecimal)
if typeof(getDecimal) != Float64
	error("getDecimal not Float64!")
end
callback("setDecimal",getDecimal)

println("getDecimalVector")
if isdefined(Main, :getDecimalVector) && !isnothing(getDecimalVector)
	error("getDecimalVector already defined!")
end
getDecimalVector = callback("getDecimalVector")
println(eltype(getDecimalVector))
println(getDecimalVector)
if eltype(getDecimalVector) != Float64
	error("getDecimalVector not Float64!")
end
callback("setDecimalVector",getDecimalVector)

println("getDecimalVectorAsList")
if isdefined(Main, :getDecimalVectorAsList) && !isnothing(getDecimalVectorAsList)
	error("getDecimalVectorAsList already defined!")
end
getDecimalVectorAsList = callback("getDecimalVectorAsList")
println(eltype(getDecimalVectorAsList))
println(getDecimalVectorAsList)
if eltype(getDecimalVectorAsList) != Float64
	error("getDecimalVectorAsList not Float64!")
end
callback("setDecimalVectorAsList",getDecimalVectorAsList)

println("getDecimalMatrix")
if isdefined(Main, :getDecimalMatrix) && !isnothing(getDecimalMatrix)
	error("getDecimalMatrix already defined!")
end
getDecimalMatrix = callback("getDecimalMatrix")
println(eltype(getDecimalMatrix))
println(getDecimalMatrix)
if eltype(getDecimalMatrix) != Float64
	error("getDecimalMatrix not Float64!")
end
callback("setDecimalMatrix",getDecimalMatrix)

println("getDecimalMatrixAsList")
if isdefined(Main, :getDecimalMatrixAsList) && !isnothing(getDecimalMatrixAsList)
	error("getDecimalMatrixAsList already defined!")
end
getDecimalMatrixAsList = callback("getDecimalMatrixAsList")
println(eltype(getDecimalMatrixAsList))
println(getDecimalMatrixAsList)
if eltype(getDecimalMatrixAsList) != Float64
	error("getDecimalMatrixAsList not Float64!")
end
callback("setDecimalMatrixAsList",getDecimalMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestDecimal.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getDecimal")
if(getDecimal ~= nil) then
	error("getDecimal already defined!")
end
getDecimal = callback("getDecimal")
getDecimalType = type(getDecimal)
print(getDecimalType)
print(getDecimal)
if(getDecimalType ~= "number") then
	error("getDecimal not Double!")
end
callback("setDecimal", getDecimal)

print("getDecimalVector")
if(getDecimalVector ~= nil) then
	error("getDecimalVector already defined!")
end
getDecimalVector = callback("getDecimalVector")
getDecimalVectorType = type(getDecimalVector[1])
print(getDecimalVectorType)
print(arraysToString(getDecimalVector))
if(getDecimalVectorType ~= "number") then
	error("getDecimalVector not Double!")
end
callback("setDecimalVector", getDecimalVector)

print("getDecimalVectorAsList")
if(getDecimalVectorAsList ~= nil) then
	error("getDecimalVectorAsList already defined!")
end
getDecimalVectorAsList = callback("getDecimalVectorAsList")
getDecimalVectorAsListType = type(getDecimalVectorAsList[1])
print(getDecimalVectorAsListType)
print(arraysToString(getDecimalVectorAsList))
if(getDecimalVectorAsListType ~= "number") then
	error("getDecimalVectorAsList not Double!")
end
callback("setDecimalVectorAsList", getDecimalVectorAsList)

print("getDecimalMatrix")
if(getDecimalMatrix ~= nil) then
	error("getDecimalMatrix already defined!")
end
getDecimalMatrix = callback("getDecimalMatrix")
getDecimalMatrixType = type(getDecimalMatrix[1][1])
print(getDecimalMatrixType)
print(arraysToString(getDecimalMatrix))
if(getDecimalMatrixType ~= "number") then
	error("getDecimalMatrix not Double!")
end
callback("setDecimalMatrix", getDecimalMatrix)

print("getDecimalMatrixAsList")
if(getDecimalMatrixAsList ~= nil) then
	error("getDecimalMatrixAsList already defined!")
end
getDecimalMatrixAsList = callback("getDecimalMatrixAsList")
getDecimalMatrixAsListType = type(getDecimalMatrixAsList[1][1])
print(getDecimalMatrixAsListType)
print(arraysToString(getDecimalMatrixAsList))
if(getDecimalMatrixAsListType ~= "number") then
	error("getDecimalMatrixAsList not Double!")
end
callback("setDecimalMatrixAsList", getDecimalMatrixAsList)
